geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/EventTimeFeatureCache.scala

```diff
@@ -0,0 +1,97 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.index
+
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+import org.geotools.util.Converters
+import org.locationtech.geomesa.filter.index.SpatialIndexSupport
+import org.locationtech.geomesa.kafka.index.KafkaFeatureCache.BasicFeatureCache
+import org.opengis.feature.simple.SimpleFeature
+import org.opengis.filter.expression.Expression
+
+import scala.util.control.NonFatal
+
+class EventTimeFeatureCache(geom: Int, dtg: Expression, dtgOrdering: Boolean, expiry: Long, support: SpatialIndexSupport)
+    extends BasicFeatureCache(geom, expiry, support) {
+
+  override protected def time: Option[Expression] = Some(dtg)
+
+  /**
+    * WARNING: this method is not thread-safe
+    *
+    * TODO: https://geomesa.atlassian.net/browse/GEOMESA-1409
```

Comment:
?!?!?!  Is this method called concurrently?

Reply:
it's not thread safe because it accesses the cache first, then the spatial index. if the cache or index happens to change in between those ops, it could cause an issue. however, while we do call this method concurrently, our consumers will process the messages for a single feature ID serially, so we shouldn't ever have threading problems. if this class is used somewhere else, it could be an issue.